components/nexusoperations/statemachine.go

```diff
@@ -500,6 +572,25 @@ func (c Cancelation) output(node *hsm.Node) (hsm.TransitionOutput, error) {
 	return hsm.TransitionOutput{Tasks: tasks}, nil
 }
 
+// TODO: Remove this implementation once transition history is fully implemented.
+func (c Cancelation) progress() (int, int32, error) {
+	switch c.State() {
```

Comment:
Note to reviewers, we never transition the timed out state the linter is complaining about.

I will double check that we're deleting the cancelation machine when the operation concludes though.

Reply:
Hmm... we never implemented deletion, so I'll just add a check in the cancelation machine to see if it still needs to be running and transition to an ABORTED state.

Reply:
But what we have for now is good enough (dropping the task):

```
		if !TransitionCanceled.Possible(op) {
			// Operation is already in a terminal state.
			return fmt.Errorf("%w: operation already in terminal state", consts.ErrStaleReference)
		}
```

Reply:
why TIMED_OUT state is defined if never transition into that state?

Reply:
It was an oversight by me.